### PR TITLE
Update the monitoring stack to terraform 0.12; remove outdated infra

### PR DESCRIPTION
**Pull request by alexwlchan**
Requires https://github.com/wellcomecollection/terraform-aws-ecs-service/pull/7; closes https://github.com/wellcomecollection/platform/issues/4363

This gets the monitoring stack running cleanly in terraform 0.12, by either inlining modules or deleting Lambdas that we aren't using any more.

- [x] Use a released version of the tf-modules

### CODE REVIEW

File: monitoring/terraform/modules/lambda/variables.tf [hunk 1]
```diff
@@ -0,0 +1,48 @@
+variable "name" {
+  description = "Name of the Lambda"
+}
+
+variable "module_name" {
+  description = "Name of the python module where the handler function lives"
+  default     = ""
```
Comment (jamesgorrie) [inline monitoring/terraform/modules/lambda/variables.tf]: Hmm?

File: monitoring/terraform/modules/lambda/variables.tf [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+variable "s3_bucket" {
+  description = "The S3 bucket containing the function's deployment package"
+}
+
+variable "s3_key" {
+  description = "The S3 key of the function's deployment package"
+}
+
+variable "memory_size" {
+  default = 128
```
Comment (jamesgorrie) [inline monitoring/terraform/modules/lambda/variables.tf]:
```suggestion
  default = 128
  type    = number 
```

File: monitoring/terraform/stack/grafana/load_balancer.tf [hunk 3]
```diff
@@ -1,29 +1,47 @@
...
+
+  target_type = "ip"
+
+  protocol = "HTTP"
+  port     = local.container_port
+  vpc_id   = var.vpc_id
+
+  health_check {
+    protocol = "HTTP"
+    path     = "/api/health"
```
Comment (jamesgorrie) [inline monitoring/terraform/stack/grafana/load_balancer.tf]: I can't see this in the previous tf, will it pass?

Comment (jamesgorrie) [reply monitoring/terraform/stack/grafana/load_balancer.tf]: [Wait, just seen it](https://github.com/wellcomecollection/platform/pull/4364/files#diff-e775a6b9b2fddd9eb9b4c0c846574c22)

File: monitoring/terraform/stack/grafana/security_groups.tf [hunk 4]
```diff
@@ -59,7 +59,7 @@ resource "aws_security_group" "external_lb_security_group" {
     cidr_blocks = ["0.0.0.0/0"]
   }
 
-  tags {
+  tags = {
```
Comment (jamesgorrie) [inline monitoring/terraform/stack/grafana/security_groups.tf]: I find these tags very strange.

File: monitoring/terraform/stack/grafana/load_balancer.tf (hunk 3 above)
Comment (alexwlchan) [reply monitoring/terraform/stack/grafana/load_balancer.tf]: It’s copied directly from the previous TF.

File: monitoring/terraform/stack/post_to_slack.tf [hunk 5]
```diff
@@ -1,69 +1,69 @@
...
   principal     = "sns.amazonaws.com"
-  source_arn    = "${var.cloudfront_errors_topic_arn}"
-  depends_on    = ["aws_sns_topic_subscription.subscribe_lambda_to_cloudfront_errors"]
+  source_arn    = var.cloudfront_errors_topic_arn
+  depends_on    = [aws_sns_topic_subscription.subscribe_lambda_to_cloudfront_errors]
 }
 
 resource "aws_sns_topic_subscription" "subscribe_lambda_to_cloudfront_errors" {
-  provider = "aws.us_east_1"
+  provider = aws.us_east_1
```
Comment (jamesgorrie) [inline monitoring/terraform/stack/post_to_slack.tf]: What's the reason for these living in Virginia?

Comment (alexwlchan) [reply monitoring/terraform/stack/post_to_slack.tf]: CloudFront lives in Virginia.

Comment (kenoir) [reply monitoring/terraform/stack/post_to_slack.tf]: CloudFront.

**Pull request merged**